pybaseball/split_stats.py

```diff
@@ -0,0 +1,129 @@
+import requests
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+from .datahelpers import postprocessing
+import bs4 as bs
+import pandas as pd
+import re
+
+
+def download_url(url: str) -> bytes:
+    """
+    Gets the content from the url specified
+    """
+    session = requests.Session()
+    retry = Retry(connect=3, backoff_factor=0.5)
+    adapter = HTTPAdapter(max_retries=retry)
+    session.mount('http://', adapter)
+    session.mount('https://', adapter)
+    resp = session.get(url)
+    return resp.content
+
+
+def get_split_soup(playerid: str, year: int = None, pitching_splits: bool = False) -> bs.BeautifulSoup:
+    """
+    gets soup for the player splits. 
+    """
+    pitch_or_bat = 'p' if pitching_splits else 'b'
+    if year is None:  # provides scores from yesterday if date is not provided
+        url = f"https://www.baseball-reference.com/players/split.fcgi?id={playerid}&year=Career&t={pitch_or_bat}"
+    else:
+        year = str(year)
+        url = f"https://www.baseball-reference.com/players/split.fcgi?id={playerid}&year={year}&t={pitch_or_bat}"
+    html = download_url(url)
+    soup = bs.BeautifulSoup(html, 'lxml')
+    return soup
+
+
+def get_player_info(playerid: str = None, soup=None) -> dict:
```

Comment:
To do these make sure you have `from typing import Dict, Optional` at the top:

```suggestion
def get_player_info(playerid: Optional[str] = None, soup: bs.BeautifulSoup = None) -> Dict:
```

Reply:
Also you say playerID can be None, but get_split_soup does not. If it's really optional, please make them agree. If it's not, please remove the None default (and don't make it `Optional[str]`, leave it as `str`)